synapse/types/__init__.py

```diff
@@ -38,7 +39,7 @@
 from immutabledict import immutabledict
 from signedjson.key import decode_verify_key_bytes
 from signedjson.types import VerifyKey
-from typing_extensions import Final, TypedDict
+from typing_extensions import TypedDict
```

Comment:
FWIW TypedDict was new in https://docs.python.org/3/library/typing.html?highlight=typeddict#typing.TypedDict , though there are various changes that were made in later Python releases